packages/web/src/components/inbox-unavailable-modal/InboxUnavailableModal.tsx

```diff
@@ -0,0 +1,179 @@
+import { ReactNode, useCallback } from 'react'
+
+import {
+  ChatPermissionAction,
+  User,
+  chatActions,
+  chatSelectors,
+  tippingActions
+} from '@audius/common'
+import {
+  IconMessageLocked,
+  IconTipping,
+  ModalTitle,
+  Modal,
+  ModalHeader,
+  Button,
+  ModalContent,
+  ModalFooter,
+  ButtonType,
+  IconUnblockMessages
+} from '@audius/stems'
+import { useDispatch } from 'react-redux'
+
+import { ArtistPopover } from 'components/artist/ArtistPopover'
+import UserBadges from 'components/user-badges/UserBadges'
+import { useGoToRoute } from 'hooks/useGoToRoute'
+import { useSelector } from 'utils/reducer'
+import { profilePage } from 'utils/route'
+
+import styles from './InboxUnavailableModal.module.css'
+
+const { unblockUser, createChat } = chatActions
+
+const messages = {
+  title: 'Inbox Unavailable',
+  content: 'You can no longer send messages to this person.',
+  button: 'Learn More',
+  tipContent: (displayName: ReactNode) => (
+    <>
+      {'You must send '}
+      {displayName}
+      {' a tip before you can send them messages.'}
+    </>
+  ),
+  tipButton: 'Send $AUDIO',
+  unblockContent: 'You cannot send messages to users you have blocked.',
+  unblockButton: 'Unblock'
+}
+
+const actionToContent = (
+  action: ChatPermissionAction,
+  user: User,
+  onClose: () => void
+) => {
+  switch (action) {
+    case ChatPermissionAction.NONE:
+      return {
+        content: messages.content,
+        buttonText: messages.button,
+        buttonIcon: null
+      }
+    case ChatPermissionAction.TIP:
+      return {
+        content: messages.tipContent(
+          <UserNameAndBadges user={user} onClickArtistName={onClose} />
+        ),
+        buttonText: messages.tipButton,
+        buttonIcon: <IconTipping />
+      }
+    case ChatPermissionAction.UNBLOCK:
+      return {
+        content: messages.unblockContent,
+        buttonText: messages.unblockButton,
+        buttonIcon: <IconUnblockMessages />
+      }
+    default:
+      return {
+        content: messages.content,
+        buttonText: messages.button,
+        buttonIcon: null
+      }
+  }
+}
+
+const { beginTip } = tippingActions
+const { getCanChat } = chatSelectors
+
+const UserNameAndBadges = ({
+  user,
+  onClickArtistName
+}: {
+  user: User
+  onClickArtistName?: () => void
+}) => {
+  const goToRoute = useGoToRoute()
+  const goToProfile = useCallback(() => {
+    goToRoute(profilePage(user.handle))
```

Comment:
Is it a bit weird to have this be clickable to navigate to the user's profile, when we're already on the profile page?

Reply:
Yeah.. but I think we should keep for consistency